data_structures/binary_tree/binary_search_tree.py

```diff
@@ -20,6 +77,12 @@ def __repr__(self) -> str:
             return str(self.value)
         return pformat({f"{self.value}": (self.left, self.right)}, indent=1)
 
+    @property
+    def is_right(self) -> bool:
+        if self.parent and self.parent.right:
+            return self == self.parent.right
+        return False
```

Comment:
I think this function can be simplified like this:

```suggestion
        return self.parent and self is self.parent.right
```

1) `==` checks if two variables have the same value, while `is` checks if they refer to the same object, so I think `is` is more appropriate here.

2) We don't need to check if `self.parent.right` is not None: if it _is_ None, then `self is self.parent.right` would fail and the function would return False anyway.